github/github.go

```diff
@@ -241,13 +241,14 @@ func NewClient(httpClient *http.Client) *Client {
 // specified, the value pointed to by body is JSON encoded and included as the
 // request body.
 func (c *Client) NewRequest(method, urlStr string, body interface{}) (*http.Request, error) {
-	rel, err := url.Parse(urlStr)
+	if c.BaseURL.Path != "" && !strings.HasSuffix(c.BaseURL.Path, "/") {
+		return nil, fmt.Errorf("BaseURL must have a trailing slash, but %q does not", c.BaseURL)
```

Comment:
I appreciate that you were thoughtful and considered the `c.BaseURL.Path` being empty (i.e., a URL like `https://api.example.com`) case. You're also handling it correctly, because for root URLs, the trailing slash is actually not neccessary. (I've confirmed, it works correctly.)

However, I will suggest we don't make this special case treatment of the root path, even though it's possible.

Removing it has the following advantages:

- It makes the API simpler; for every type of URL, it must always have a trailing slash. This is simpler to communicate than "all URLs must have a trailing slash, except if the path is root, then it's optional."
- It eliminates ambiguity of whether the trailing slash should be included for root URLs. It would always need to be included.
- It makes the code simpler (less special cases).

So, my suggestion is to do this:

```Go
if !strings.HasSuffix(c.BaseURL.Path, "/") {
	return nil, fmt.Errorf("BaseURL must have a trailing slash, but %q does not", c.BaseURL)
}
```

That'll mean root URLs will need to be provided as `https://api.example.com/` instead of `https://api.example.com`, which is completely okay.

What do you think?